pip_cache/__init__.py

```diff
@@ -9,38 +9,66 @@
 from __future__ import print_function
 import os
 import sys
+
+import marisa_trie
+
 from .xdg import get_xdg_data_dir
 import argparse
 
-#TODO speed up by using msgpack to store a marisa-trie for quick prefix lookup
-# I haven't had the need to do this yet, since it's fast enough for autocomplete
-# already
-#import marisa-trie
-#import msgpack
-pip_cache_data_dir = os.path.join(get_xdg_data_dir(), 'pip-cache')
-index_filename = os.path.join(pip_cache_data_dir, 'all-packages.txt')
 
-# make sure that the directory we might be writing in exists
-if not os.path.isdir(pip_cache_data_dir):
-    os.mkdir(pip_cache_data_dir)
+def get_pip_cache_data_dir():
+    return os.path.join(get_xdg_data_dir(), 'pip-cache')
+
+
+def get_raw_index_filename():
+    return os.path.join(get_pip_cache_data_dir(), 'all-packages.txt')
+
+
+def get_index_filename():
+    return os.path.join(get_pip_cache_data_dir(), 'all-packages.marisa')
```

Comment:
FYI: I changed these to methods, instead of global constants, so I could easily patch them out during testing